src/assets/js/react/components/FontManager/FontList.js

```diff
@@ -1,46 +1,43 @@
-import React, { Component } from 'react'
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
 import FontListHeader from './FontListHeader'
-import FontListItem from './FontListItem'
-
-export class FontList extends Component {
-  state = {
-    items: [
-      1,
-      2,
-      3,
-      4,
-      5,
-      6,
-      7,
-      8,
-      9,
-      10,
-      11,
-      12,
-      13,
-      14,
-      15,
-      16,
-      17,
-      18,
-      19,
-      20
-    ]
-  }
-
-  render () {
-    return (
-      <div className='font-list'>
-        <FontListHeader />
-
-        <div className='font-list-items'>
-          {
-            this.state.items.map(item => <FontListItem key={item} />)
-          }
-        </div>
-      </div>
-    )
-  }
+import FontListItems from './FontListItems'
+import FontListSkeleton from './FontListSkeleton'
+import FontListAlertMessage from './FontListAlertMessage'
+
+const FontList = ({ id, loading, fontList, searchResult, msg: { error }, history }) => {
+  const fontListError = error && error.fontList
+  const checkSearchResult = (searchResult && searchResult.length === 0) || !searchResult
+  const latestData = fontList.length > 0 && !searchResult
+
+  return (
+    <div className='font-list'>
+      <FontListHeader />
+
+      {loading ? <FontListSkeleton /> : <FontListItems id={id} history={history} />}
+
+      {(!fontListError && !loading) && !latestData && checkSearchResult && <FontListAlertMessage />}
```

Comment:
That's a lot of inline checks. Best move it above the return statement and save it to its own `const`